File: trunk/20102/tr/proposta01/Factorial.php
<?php

class Factorial
{
    private $input;

    public function __construct($input)
    {
        $this->input = $input;
    }

    public function execute()
    {
        $answer = 1;
        $x = 1;
        while ($x <= $this->input) {
            $answer = $answer * $x;
            $x = $x + 1;
        }
        return $answer;
    }
}

$f = new Factorial(10);
$result = $f->execute();

File: 20102/tr/proposta01/article/Factorial.java
public class Factorial
{
    private int input;

    public Factorial(int input)
    {
        this.input = input;
    }

    public int execute()
    {
        int answer = 1;
        int x = 1;
        while (x <= input) {
            answer = answer * x;
            x = x + 1;
        }
        return answer;
    }

    public static void main(String args[])
    {
        Factorial f = new Factorial(10);
        int result = f.execute();
    }
}